packages/opentelemetry-exporter-stackdriver-trace/src/transform.ts

```diff
@@ -0,0 +1,130 @@
+/*!
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { hrTimeToTimeStamp, VERSION } from '@opentelemetry/core';
+import { ReadableSpan } from '@opentelemetry/tracing';
+import * as ot from '@opentelemetry/types';
+import {
+  AttributeMap,
+  Attributes,
+  AttributeValue,
+  Link,
+  LinkType,
+  Span,
+  TruncatableString,
+} from './types';
+
+const AGENT_LABEL_KEY = 'g.co/agent';
+const AGENT_LABEL_VALUE = `opentelemetry-version [${VERSION}]`;
+
+export function getReadableSpanTransformer(
+  projectId: string,
+  serviceName: string
+): (span: ReadableSpan) => Span {
+  return span => {
+    const attributes = transformAttributes(span.attributes, {
+      project_id: projectId,
+      service: serviceName,
+      [AGENT_LABEL_KEY]: AGENT_LABEL_VALUE,
+    });
+
+    const out: Span = {
+      attributes,
+      displayName: stringToTruncatableString(span.name),
+      links: {
+        link: span.links.map(transformLink),
+      },
+      endTime: hrTimeToTimeStamp(span.endTime),
+      startTime: hrTimeToTimeStamp(span.startTime),
+      name: `projects/${projectId}/traces/${span.spanContext.traceId}/spans/${span.spanContext.spanId}`,
+      spanId: span.spanContext.spanId,
+      status: span.status,
+      timeEvents: {
+        timeEvent: span.events.map(e => ({
+          time: hrTimeToTimeStamp(e.time),
+          annotation: {
+            attributes: e.attributes,
+            description: stringToTruncatableString(e.name),
+          },
+        })),
+      },
+    };
+
+    if (span.parentSpanId) {
+      out.parentSpanId = span.parentSpanId;
+    }
+
+    return out;
+  };
+}
+
+function transformLink(link: ot.Link): Link {
+  return {
+    attributes: transformAttributes(link.attributes),
+    spanId: link.spanContext.spanId,
+    traceId: link.spanContext.traceId,
+    type: LinkType.UNSPECIFIED,
+  };
+}
+
+function transformAttributes(
+  requestAttributes: ot.Attributes = {},
+  serviceAttributes: ot.Attributes = {}
+): Attributes {
+  const attributes = Object.assign({}, requestAttributes, serviceAttributes);
+  const attributeMap = transformAttributeValues(attributes);
+  return {
+    attributeMap: attributeMap,
+    droppedAttributesCount:
+      Object.keys(attributes || {}).length - Object.keys(attributeMap).length,
```

Comment:
I think this is not right, ideally we should capture value from here => https://github.com/open-telemetry/opentelemetry-js/blob/master/packages/opentelemetry-tracing/src/Span.ts#L86-L93 and use it here. Maybe, just add `@todo` and assign `0` for now.

Reply:
We may need both. Right now we are also dropping values which are not `string | number | boolean` because SD only supports those values. WDYT

Reply:
True, we need both the values. Maybe, just add `@todo` to get the value from the SDK via `ReadableSpan`.